core/unit_test/TestReduceCombinatorical.hpp

```diff
@@ -414,10 +414,14 @@ struct TestReduceCombinatoricalInstantiation {
 #endif
   }
 
+  struct AddIndexToSum {
```

Comment:
I would prefer if we guarded that change for CUDA RDC enabled and annotated with a FIXME or something clearly indicating that we expect this is a toolchain bug.

Reply:
Not just an RDC problem. I reproduced with
```
cd $BUILD_DIR
cmake -DCMAKE_BUILD_TYPE:STRING=Release \
      -DCMAKE_CXX_COMPILER=$KOKKOS_DIR/bin/nvcc_wrapper \
      -DCMAKE_CXX_STANDARD=17 \
      -DKokkos_ENABLE_OPENMP=OFF \
      -DKokkos_ENABLE_SERIAL=ON \
      -DKokkos_ENABLE_CUDA=ON \
      -DKokkos_ENABLE_CUDA_RELOCATABLE_DEVICE_CODE=OFF \
      -DKokkos_ENABLE_TESTS=ON \
      $KOKKOS_DIR

cd $BUILD_DIR/core/unit_tests
make -j 20
./Kokkos_CoreUnitTest_Default --gtest_shuffle --gtest_filter=*reduce_instantiation_*
```
And I've reproduced with cuda11 and cuda12.

Reply:
OK let's add a big fat FIXME_CUDA comment then